File: python/life/life/cli_display.py
# import subprocess
# import sys
# from typing import TextIO


# window: TextIO = None


def render(frame, new_window=False, refresh=False):
    image = [''.join(['██' if element else '  ' for element in row]) for row in frame]
    image = '\n'.join(image)

    # global window
    # if window is None:
    #     window = TextIO() if new_window else sys.stdout
    # if refresh:
    #     print('\f')  # version 1 - doesn't clear buffer
    # # window.writelines(image)
    # window.write(image)
    # window.flush()

    print(image)

File: rust/life/src/cli_display.rs
use std::process::Command;

pub fn render(frame: Vec<Vec<u8>>, refresh: Option<bool>) {
    let image = frame
        .iter()
        .map(|vec| vec
            .iter()
            .map(|&state| if state == 1u8 {"██"} else {"  "})
            .collect::<Vec<&str>>()
            .join(""))
        .collect::<Vec<String>>()
        .join("\n");
    
    if refresh.unwrap_or_default() {
        Command::new("clear");
    }

    print!("{}", image);
}